server/aws-lsp-codewhisperer/src/language-server/telemetryService.ts

```diff
@@ -52,7 +54,24 @@ export class TelemetryService extends CodeWhispererServiceToken {
         return suggestionState
     }
 
+    private isLoginInvalidForTelemetry(): boolean {
+        return this.loginType === 'iam' || (this.loginType !== 'identityCenter' && this.optOutPreference !== 'OPTOUT')
+    }
+
+    private invokeSendTelemetryEvent(event: any) {
```

Comment:
nit: can we specify type instead of `any` for `event`?

Reply:
I added any intentionally because this is like a common method I intend to use for all the events. Once, all are implemented - I'll change it to Event1|Event2|.. something like this.